nixos/modules/programs/xppen.nix

```
@@ -0,0 +1,27 @@
+{ config, lib, pkgs, ... }:
+
+with lib; let
+  cfg = config.programs.xppen;
+in
+
+{
+
+  options.programs.xppen = {
+    package = mkPackageOption pkgs "xppen" { };
+
+    enable = mkEnableOption (lib.mdDoc "XPPen PenTablet application");
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [
+      cfg.package
+    ];
+
+    services.udev.extraRules = ''
+      # XPPen PenTablet rules
+      KERNEL=="uinput",MODE:="0666",OPTIONS+="static_node=uinput"
+      SUBSYSTEMS=="usb",ATTRS{idVendor}=="28bd",MODE:="0666"
+    '';
```

Maybe these rules could be packaged into the package itself? I think it would be useful for anyone who's not using NixOS

---

It doesn't automatically activate them unfortunately, you'll still need the module for that. But other people using nixpkgs for non-nixos systems will benefit from having these sorts of files available. Exactly how they use nixpkgs, and how they install these files is up to them.

If you install it in something like `$out/lib/udev.d/rules/50-xppen.rules`, this can be rewritten as:

```suggestion
    services.udev.packages = [ cfg.package ];
```